azblob/zc_sas_query_params.go

```diff
@@ -200,11 +181,11 @@ func newSASQueryParameters(values url.Values, deleteSASParametersFromValues bool
 		case "spr":
 			p.protocol = SASProtocol(val)
 		case "snapshot":
-			p.snapshotTime, _ = time.Parse(SnapshotTimeFormat, val)
+			p.snapshotTime = val
 		case "st":
```

Comment:
Just a note here, but in the Files SDK we can parse multiple time formats, I believe.

Something similar may be beneficial here.

Reply:
Yes, I called that change out earlier today I think.  I still don't like that it's trying to parse a `Time` out from the string, even if using multiple format strings, because even if we retain the string and send it back out, the `Time` value itself may not match and could lead to confusion.  But I see the value in letting the API maintain a `Time` too.